### ISSUE TITLE
[Enhancement] When there are too many consumers and there are too many RocketMQ client threads

### ISSUE THREAD

**Issue by imgoby** [apache/rocketmq#9035]
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

When there are too many consumers and there are too many RocketMQ client threads, details:
NettyClientWorkerThread_1: 4746 threads
Client House Keeping Service: 2373 threads
CleanExpireMsgSchechedThread_: 4378 threads
RebalanceServic: 2373 threads
NettyClientSele: 2373 threads

How to resolve?

### Motivation

reduce threads, reduce memory.

### Describe the Solution You'd Like

Merge timers, schedule executor

### Describe Alternatives You've Considered

Merge timers, schedule executor

**Comment by yx9o** [comment]
Hi, I want to know how the client is used. Will multiple consumers be created in the same process?

**Comment by imgoby** [comment]
yes. in the same process.the process is a java application using springboot+rocketmq
For example, there is a topic: hello_topic， Then there were 3000 consumers who consumed the topic according to the conditions and found that the process memory occupied 10G

**Comment by yx9o** [comment]
Is `instanceName` set?

**Comment by imgoby** [comment]
No, can this parameter solve it?

**Comment by Yelijah** [comment]
Rocketmq grpc client will create a thread pool for every consumer or producer.  So I  modify the client source code to fix this problem。

**Comment by imgoby** [comment]
which version?

**Comment by chenchengwangS** [comment]
![Image](https://github.com/user-attachments/assets/8112f025-8ae1-44ab-ab52-74fe4ab2b955)
